src/rich_click/__init__.py

```diff
@@ -32,22 +37,93 @@
         "version_option",
         "group",
         "command",
+        "rich_config",
+        "RichContext",
+        "RichHelpConfiguration",
     ]
 
 
-def group(*args, cls=RichGroup, **kwargs):
+def group(name=None, cls=RichGroup, **attrs) -> Callable[..., RichGroup]:
     """
     Group decorator function.
 
     Defines the group() function so that it uses the RichGroup class by default.
     """
-    return click_group(*args, cls=cls, **kwargs)
 
+    def wrapper(fn):
+        if hasattr(fn, "__rich_context_settings__"):
+            rich_context_settings = getattr(fn, "__rich_context_settings__", {})
+            console = rich_context_settings.get("rich_console", None)
+            help_config = rich_context_settings.get("help_config", None)
+            context_settings = attrs.get("context_settings", {})
+            context_settings.update(rich_console=console, rich_help_config=help_config)
+            attrs.update(context_settings=context_settings)
+            del fn.__rich_context_settings__
+        if callable(name) and cls:
+            group = click_group(cls=cls, **attrs)(name)
+        else:
+            group = click_group(name, cls=cls, **attrs)
+        cmd = cast(RichGroup, group(fn))
+        return cmd
 
-def command(*args, cls=RichCommand, **kwargs):
+    return wrapper
+
+
+def command(*args, cls=RichCommand, **kwargs) -> Callable[..., RichCommand]:
```

Comment:
@davidbrochart do we need to update `command` as well as `group` in the same way as done in #104?

Reply:
Ok, I basically copied the change you made and applied it to this function as well (modifying where relevant to be command instead of group). Hope that's ok 🤞🏻 

I was never able to replicate the error that you were fixing in the first place, so I'll leave it to you to shout if it's broken in any way again 😁

Reply:
Indeed, thanks! There was just a small typo (see #105).

Reply:
Nice catch - thanks! 🙌🏻